app/internal/translation.py

```diff
@@ -0,0 +1,77 @@
+from typing import Optional
+
+from iso639 import languages
+from sqlalchemy.exc import SQLAlchemyError
+from textblob import TextBlob, download_corpora
+from textblob.exceptions import NotTranslated
+
+from app.database.database import SessionLocal
+from app.database.models import User
+
+download_corpora.download_all()
+
+
+def translate_text(text: str,
+                   target_lang: str,
+                   original_lang: Optional[str] = None
+                   ) -> str:
+    """
+    Translate text to the target language
+    optionally given the original language
+    """
+    if len(text) <= 0:
+        return "No text to translate"
+    if original_lang is None:
+        original_lang = _detect_text_language(text)
+    else:
+        original_lang = _lang_full_to_short(original_lang)
+
+    if original_lang == _lang_full_to_short(target_lang):
+        return text
+
+    try:
+        return str(TextBlob(text).translate(
+            from_lang=original_lang,
+            to=_lang_full_to_short(target_lang)))
+    except NotTranslated:
+        return text
+
+
+def _detect_text_language(text: str) -> str:
+    """
+    Gets some text and returns the language it is in
+    Uses external API
+    """
+    return str(TextBlob(text).detect_language())
+
+
+def _get_user_language(user_id: int, session: SessionLocal) -> str:
+    """
+    Gets a user-id and returns the language he speaks
+    Uses the DB"""
+    try:
+        user = (
+            session.query(User).filter(User.id == user_id).first()
+        )
+        language_user = user.language
+    except SQLAlchemyError:
+        return ""
+    else:
+        return language_user
+
+
+def translate_text_for_user(text: str,
+                            session: SessionLocal,
+                            user_id: int) -> str:
+    """
+    Gets a text and a user-id and returns the text,
+    translated to the language the user speaks
+    """
+    target_lang = _get_user_language(user_id, session)
+    if not target_lang:
+        return text
+    return translate_text(text, target_lang)
+
+
+def _lang_full_to_short(full_lang: str) -> str:
+    return languages.get(name=full_lang.capitalize()).alpha2
```

Comment:
What is alpha2? :)

Reply:
it is language name using two letters.
english -> en
french -> fr

Reply:
Nice, I think it will be a good idea to add short documentation for this (even in a single, short sentence)

Reply:
oh, I missed that when I was writing the docs. Thank you, now I will add :)